src/Microsoft.Azure.WebJobs.Extensions.Kafka/Output/KafkaProducerAsyncCollector.cs

```diff
@@ -78,13 +78,39 @@ private object BuildKafkaDataEvent(T item)
             return new KafkaEventData<T>(item);
         }
 
-        private KafkaEventData<string> BuildKafkaEventData(JObject dataObj)
+        private object BuildKafkaEventData(JObject dataObj)
+        {
+            if (dataObj["Key"] != null && (string)dataObj["Key"] != null)
```

Comment:
Why are you using && for the same object? (looks different cast)?

Reply:
Done